src/state/slices/opportunitiesSlice/resolvers/idle.ts

```diff
@@ -0,0 +1,100 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import { fromAssetId, toAssetId } from '@shapeshiftoss/caip'
+import { DefiProvider, DefiType } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import { getIdleInvestor } from 'features/defi/contexts/IdleProvider/idleInvestorSingleton'
+import { logger } from 'lib/logger'
+import { selectAssetById } from 'state/slices/selectors'
+
+import type {
+  GetOpportunityIdsOutput,
+  GetOpportunityMetadataOutput,
+  OpportunitiesState,
+  OpportunityDefiType,
+  StakingId,
+} from '../types'
+import type { ReduxApi } from './types'
+
+const moduleLogger = logger.child({ namespace: ['opportunities', 'resolvers', 'idle'] })
+
+export const idleStakingOpportunitiesMetadataResolver = async ({
+  opportunityType,
+  reduxApi,
+}: {
+  opportunityType: OpportunityDefiType
+  reduxApi: ReduxApi
```

Comment:
Type me :)

Reply:
Will stack a PR on top of this so we tackle it for `foxFarming` resolvers as well without cluttering this diff with it